enhancements/bpfd-on-ocp.md

```diff
@@ -0,0 +1,424 @@
+
+4. As an OCP Cluster Admin, I want to know ALL of the BPF programs deployed in
+   my Cluster.
+
+   As more and more kubernetes applications utilize BPF, cluster stability may
+   ultimately be threatened. In order to mitigate this risk admins should be
+   able to easily see all of the BPF programs loaded on a given cluster.
+
+5. As an OCP Developer or support engineer I would like to easily utilize
+   advanced BPF debugging tools. 
+
+   Tools such as bpftrace provide unprecedented insight into the inner workings
+   of the kernel, however, support for deploying and running those tools in OCP
+   is very limited. 
+
+6. As an OCP Developer, I need a centralized way to package and deploy BPF
+   applications that follow secure supply chain best practices.
+
+   Specifically, it should be simple for me to have fine-grained control over
+   the versioning of both the user space and kernel space portions of my BPF
+   applications. Additionally, shipping changes to both portions of the
+   application should be straightforward and make use of existing
+   infrastructure, e.g container registries. Lastly, I need to follow supply
+   chain security best practices with BPF, e.g bytecode signing and integration
+   with Sigstore
+
+7. As an OCP Admin, I want to apply policy around BPF to control which users and
+   applications can/cannot perform BPF operations. 
+
+   Some BPF programs are critical to a functioning datapath while others may
+   simply report data back to the user. Therefore policy regarding the actors
+   who can load such programs should be explicit and easily integrated with OCP
+   RBAC. 
+
+8. As an OCP Developer/Admin I want to limit the container permissions required
+   to load and run BPF programs. 
+
+   In OCP today loading and managing BPF programs often requires that the pod be
+   privileged and have both the CAP_BPF and CAP_NET_ADMIN capabilities. This is
+   ultimately a major security concern and should be mitigated to ensure
+   applications can load BPF programs without such privileges.
+
+### Goals
+
+The goal of this proposal is to provide OCP developers and customers with a
+consistent and more secure experience using BPF. Whether they are consuming BPF
+as a component part of OCP, from a partner product, or producing BPF themselves.
+
+If we succeed in our goal, we will provide a solution, or at least the
+infrastructure for a solution, for the user stories listed above. In particular:
+
+- It will be simpler for OCP developers to use BPF in their work.
+- Red Hat teams can collaborate on BPF development.
+- OCP administrators will be able to easily discover what BPF has been loaded in
+  their cluster, who loaded it, and for what purpose.
+- Security will be improved because applications leveraging BPF will not need to
+  run in privileged containers.
+
+### Non-Goals
+
+We do not want to mandate that customers ONLY use our method of working with BPF
+in OpenShift.
```

Comment:
We're pretty opinionated about how they configure their OS in other ways. Why not set some support boundaries in this case, too?

How well would bpfd coexist with something else?

Reply:
I think this question should be open for discussion.  It was not our intent to mandate it.  As long as you have a privileged container with CAP_BPF, you can load BPF programs any way you like.  These programs can be a security risk and can interfere with other BPF programs.  Setting some support boundaries sounds reasonable to me.